graphql-dgs-reactive/src/main/kotlin/com/netflix/graphql/dgs/reactive/internal/DefaultDgsReactiveQueryExecutor.kt

```diff
@@ -0,0 +1,153 @@
+            Mono.fromCompletionStage(
+                BaseDgsQueryExecutor.baseExecute(
+                    query,
+                    variables,
+                    operationName,
+                    dgsContext,
+                    graphQLSchema,
+                    dataLoaderProvider,
+                    chainedInstrumentation,
+                    queryExecutionStrategy,
+                    mutationExecutionStrategy,
+                    idProvider
+                )
+            ).doOnEach { result ->
+                if (result.hasValue()) {
+                    val nullValueError = result.get()?.errors?.find { it is NonNullableFieldWasNullError }
+                    if (nullValueError != null) {
+                        logger.error(nullValueError.message)
+                    }
+                }
+            }
+        }
+    }
+
+    override fun <T : Any> executeAndExtractJsonPath(
+        query: String,
+        jsonPath: String,
+        variables: MutableMap<String, Any>
+    ): Mono<T> {
+        return getJsonResult(query, variables).map { JsonPath.read(it, jsonPath) }
+    }
+
+    override fun executeAndGetDocumentContext(
+        query: String,
+        variables: MutableMap<String, Any>
+    ): Mono<DocumentContext> {
+        return getJsonResult(query, variables).map(BaseDgsQueryExecutor.parseContext::parse)
+    }
+
+    override fun <T : Any?> executeAndExtractJsonPathAsObject(
+        query: String,
+        jsonPath: String,
+        variables: MutableMap<String, Any>,
+        clazz: Class<T>
+    ): Mono<T> {
+        return getJsonResult(query, variables)
+            .map(BaseDgsQueryExecutor.parseContext::parse)
+            .map {
+                try {
+                    it.read(jsonPath, clazz)
+                } catch (ex: MappingException) {
+                    throw DgsQueryExecutionDataExtractionException(ex, it.jsonString(), jsonPath, clazz)
+                }
+            }
+    }
+
+    override fun <T : Any?> executeAndExtractJsonPathAsObject(
+        query: String,
+        jsonPath: String,
+        variables: MutableMap<String, Any>,
+        typeRef: TypeRef<T>
+    ): Mono<T> {
+        return getJsonResult(query, variables)
+            .map(BaseDgsQueryExecutor.parseContext::parse)
+            .map {
+                try {
+                    it.read(jsonPath, typeRef)
+                } catch (ex: MappingException) {
+                    throw DgsQueryExecutionDataExtractionException(ex, it.jsonString(), jsonPath, typeRef)
+                }
+            }
+    }
+
+    private fun getJsonResult(query: String, variables: Map<String, Any>): Mono<String> {
+        return execute(query, variables).map { executionResult ->
+            if (executionResult.errors.size > 0) {
+                throw QueryException(executionResult.errors)
+            }
+
+            val objectMapper = ObjectMapper()
```

Comment:
Do we really need to create `ObjectMapper ` here every time?
Also, mb we can provide an ability to inject a custom realization?

Reply:
It's now using the objectmapper created in the `BaseDgsQueryExecutor` companion object; that way it's also using the same configuration as when using JsonPath. 

We can consider making it injectable later.